templates/cicd/pipeline_cfn.yml

```diff
@@ -235,27 +242,40 @@ Resources:
       Stages:
         - Name: Source
           Actions:
-            - Name: SourceCodeFor-{{$.AppName}}
-              ActionTypeId:
-                Category: Source
-                # TODO: #273 For CodeCommit, this needs to be AWS. Let's do that later
-                Owner: ThirdParty
-                Version: 1
-                Provider: {{.Source.ProviderName}}
-              Configuration:
-                # TODO: #273 For now this set of configurations is GitHub-specific,
-                # change it to be more generic
-                Owner: {{$.Source.Owner}}
-                Repo: {{$.Source.Repository}}
-                Branch: {{index $.Source.Properties "branch"}}
-                # https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/dynamic-references.html#dynamic-references-secretsmanager
-                # Use the *entire* SecretString with version AWSCURRENT
-                OAuthToken: !Sub
-                    - '{{"{{"}}resolve:secretsmanager:${SecretId}{{"}}"}}'
-                    - { SecretId: !Ref GitHubAccessTokenSecretId }
-              OutputArtifacts:
-                - Name: SCCheckoutArtifact
-              RunOrder: 1
+            !If
+              - IsGitHub
+              -
```

Comment:
🤔 I wonder if we should just use go templates instead of CFN conditionals for this?
```yaml
{{- if eq .Source.ProviderName "GitHub"}}
   ....
{{- end }}
{{- if eq .Source.ProviderName "CodeCommit"}}
   ....
{{- end }}
```
In a later PR, we can then move these separate source YAML blocks to a "partials/" directory and do:
```yaml
{{- if eq .Source.ProviderName "GitHub"}}
{{include "github-source" .Source | indent 12}}
{{- end }}
```

Reply:
Ah, good to know about this cool 'partials' option--I had originally been doing more conditions inline because I was worried about the template getting unwieldy with multiple source options.

Reply:
Done.